### DISCUSSION TITLE
How Do I Connect SDR++ to Gpredict?

### DISCUSSION THREAD

**Discussion by AlexLandherr** [AlexandreRouma/SDRPlusPlus#1244] [Q&A]
How do I best connect SDRpp to Gpredict so that Gpredict can control the frequency of SDR++?

My aim is to track NOAA APT satellites (15, 18 and 19).

I'm on Debian 12 Bookworm and I've successfully set up SDR++ so I can receive FM radio via my HackRF One.

**Comment by brentabget** [comment]
if you still need to know how to make the two work together 

 I will sent you a link on how to make it work .
I haft to find the link!!
I have my working and it works vary well.

**Reply by AlexLandherr** [reply]
Did you find the relevant link?

**Comment by brentabget** [comment]
https://www.youtube.com/watch?v=AghjgwTw-Vk
In sdr++
Open the tick Rigctlr Server
You will see the info you need to set Gpredict settings
Let me know if you need more help.

**Reply by AlexLandherr** [reply]
What should I set the audio record options to in SDR++? The Doppler corrections are coming through now.

**Reply by AlexLandherr** [reply]
Is there any way to make Gpredict remember the downlink frequency for each satellite when sending tracking/doppler commands to SDR++ so I don't have to enter it under "Radio Control" each time?

E.g. each time I or Gpredict switch satellite the frequency in SDR++ should also jump to the corresponding one for NOAA 15, 18 and 19.

**Comment by brentabget** [comment]
When you open Gpredict You will see your chosen (Favorites) Sats.
Those will be the Sat. you will see in Radio Control in the Tic down error.

**Comment by brentabget** [comment]
In sdrpp select recoarding and i think it will default to one of your
directorys ( Music,Documents, Downloads)
Let me know if you can't get it working, If you can't I will set it up and
see the precise directory where it stores the auto file.

**Reply by AlexLandherr** [reply]
I have already set the directory for recording and have that working; I meant what sample type and container format should be set for NOAA APT?

**Comment by brentabget** [comment]
i  would use wav and int16

I  would haft to set it up to be sure

**Comment by brentabget** [comment]
When I get in this after_noon I will set it up again and then i can tell
you exactly how to do it.

**Reply by AlexLandherr** [reply]
Might not be necessary; I think I just figured it out.

Is there any way to correct the APT frequency for NOAA-15?
The transponder data is wrong when it says 137.5 MHz when it should be 137.62 MHz per NOAA's own POES status page.

**Comment by brentabget** [comment]
Not sute about APT frequency
"The transponder data is wrong when it says 137.5 MHz when it should be
137.62 MHz per NOAA's own POES status page"

The only way they will be the same is if the Sat. is directly over head (90
Deg) up from your location.

Gpredict displats tow frequencies
1.
Transponder Freq.
2.
Doppler off set Freq.

**Reply by AlexLandherr** [reply]
As in the frequency the satellite transmits at is wrong (sorry for using the wrong terminology), if that is wrong the Doppler correction will be wrong.
I can manually adjust it but have to do it every time I tried it.

**Comment by brentabget** [comment]
In the Gpredict Radio control the Sat. Freq.  You haft to set it to  the
wright Freq by clicking the up-down errors.

**Reply by AlexLandherr** [reply]
Yes, but is there really no way to keep it from returning to 137.5 MHz and stay on the correct 137.62 MHz **without** having to manually adjust it every time?

**Reply by AlexandreRouma** [reply]
Do not use doppler correction on NOAA APT. This mode was literally engineered to not require and. All you're doing is creating jumps in the demodulated audio as the VFO is moved a few Hz at a time adding noise to the image for no reason.

**Reply by AlexLandherr** [reply]
In my experience the image gets slightly warped if there is no Doppler correction; so despite what you say it's still needed.

**Reply by AlexandreRouma** [reply]
I don't know what software you're using but anything decent will have no issue with it.

**Reply by AlexLandherr** [reply]
I've used Orbitron for tracking and sending Doppler corrections, SDR# and WXtoImg for decoding the audio stream from SDR#.

**Reply by AlexandreRouma** [reply]
So that was with the worse possible software at every step of the way. Maybe if it was 2012 this would be a decent choice but nowadays there are better performing software. Orbitron is horribly outdated but at least it works, so is WXtoImg which is an abandoned proprietary mess full of bugs and SDR# doesn't have the specialized noise reduction that SDR++ has. Use SDR++ for demod and NOAA-APT or SatDump for decoding.

**Reply by AlexLandherr** [reply]
Ok, understood.

**Comment by brentabget** [comment]
It been a while for me sense i used the radio controll
I set sdrpp to the Sat. Freq. then started the Rigctl server
Started tracking with Gpredict Radio controll and the two worked perfectly
Wait untill you get a NOAA Sat path close to you and give a try.
I think the two of them togather will suprise you how good they work.
Let me know.

**Reply by AlexLandherr** [reply]
Ok, whatever database is used to get the satellite downlink frequency should be updated to actually correspond to what NOAA says.

**Comment by brentabget** [comment]
When you decode NOAA Weather transmissions I don't think you want any kind
of noise reduction I thind you want raw data.

**Reply by AlexandreRouma** [reply]
I wrote this damn thing, I know what you're supposed to use or not use...
This noise reduction algo was designed and tested specifically for it. See the difference between on and off here:
https://www.reddit.com/media?url=https%3A%2F%2Fpreview.redd.it%2Fu3o5ctqof4781.gif%3Fformat%3Dmp4%26s%3Decc8046a673f6b5f100f815c10b63f8c6c2f370e
